query/promql/internal/promqltests/tests.go

```diff
@@ -0,0 +1,533 @@
+		},
+		{
+			// label_replace fails when the regex is invalid.
+			Query:      `label_replace(demo_num_cpus, "job", "value-$1", "src", "(.*")`,
+			ShouldFail: true,
+		},
+		{
+			// label_replace fails when the destination label name is not a valid Prometheus label name.
+			Query:      `label_replace(demo_num_cpus, "~invalid", "", "src", "(.*)")`,
+			ShouldFail: true,
+		},
+		{
+			// label_replace fails when there would be duplicated identical output label sets.
+			Query:      `label_replace(demo_num_cpus, "instance", "", "", "")`,
+			ShouldFail: true,
+		},
+		{
+			Query: `label_join(demo_num_cpus, "new_label", "-", "instance", "job")`,
+		},
+		{
+			Query: `label_join(demo_num_cpus, "job", "-", "instance", "job")`,
+		},
+		{
+			Query: `label_join(demo_num_cpus, "job", "-", "instance")`,
+		},
+		{
+			Query:      `label_join(demo_num_cpus, "~invalid", "-", "instance")`,
+			ShouldFail: true,
+		},
+		{
+			Query: `label_join(demo_num_cpus, "_value", "-", "_stop")`,
+		},
+		{
+			Query:       `{{.dateFunc}}()`,
+			VariantArgs: []string{"dateFunc"},
+		},
+		{
+			Query:       `{{.dateFunc}}(demo_batch_last_success_timestamp_seconds)`,
+			VariantArgs: []string{"dateFunc"},
+		},
+		{
+			Query:       `{{.instantRateFunc}}(demo_cpu_usage_seconds_total[{{.range}}])`,
+			VariantArgs: []string{"instantRateFunc", "range"},
+		},
+		{
+			Query:       `{{.clampFunc}}(demo_cpu_usage_seconds_total, 2)`,
+			VariantArgs: []string{"clampFunc"},
+		},
+		{
+			Query:       `resets(demo_cpu_usage_seconds_total[{{.range}}])`,
+			VariantArgs: []string{"range"},
+		},
+		{
+			Query:       `changes(demo_batch_last_success_timestamp_seconds[{{.range}}])`,
+			VariantArgs: []string{"range"},
+		},
+		{
+			Query: `vector(1)`,
+		},
+		{
+			Query: `vector(time())`,
+		},
+		{
+			Query:       `histogram_quantile({{.quantile}}, rate(demo_api_request_duration_seconds_bucket[1m]))`,
+			VariantArgs: []string{"quantile"},
+		},
+		{
+			Query: `histogram_quantile(0.9, nonexistent_metric)`,
+		},
+		{
+			// Missing "le" label.
+			Query: `histogram_quantile(0.9, demo_cpu_usage_seconds_total)`,
+		},
+		{
+			// Missing "le" label only in some series of the same grouping.
+			Query: `histogram_quantile(0.9, {__name__=~"demo_api_request_duration_seconds_.+"})`,
+		},
+		{
+			Query:       `holt_winters(demo_disk_usage_bytes[10m], {{.smoothingFactor}}, {{.trendFactor}})`,
+			VariantArgs: []string{"smoothingFactor", "trendFactor"},
+		},
+		{
+			Query:          `max_over_time((time() - max(demo_batch_last_success_timestamp_seconds) < 1000)[5m:10s] offset 5m)`,
+			SkipComparison: "the implementation cannot guarantee completely identical results for subqueries",
+		},
+		{
+			Query:          `avg_over_time(rate(demo_cpu_usage_seconds_total[1m])[2m:10s])`,
+			SkipComparison: "the implementation cannot guarantee completely identical results for subqueries",
+		},
+	}
+)
+
+func init() {
+	TestCases = getTestCases()
```

Comment:
nit: this doesn't need to be an `init`, this could just be part of the top-level var declaration and assignment. But fine to leave for this PR.